### PR TITLE
[cmake] Add cmake port to test qt 

### PR DESCRIPTION
**Pull request by Neumann-A**
PR #10769 is included for static builds to work

~~cmake-gui in static builds will only work with Qt > 5.14 due to the automatic plugin linkage added to Qt 5.14. Seems like cmake forgot to link against and add sources to include the required qt plugins.~~

- [x] waiting for the qt 5.15 merge so i don't have to deal with the plugins in static builds ;)
- [x] retesting and updating

### CODE REVIEW

**Comment by Neumann-A** [comment]
also requires at least #10644 to make static builds work

**Comment by cenit** [comment]
This pr might also help you a lot with qt maintenance. It will also make sure that no one will break it bigly and involuntarily

**Comment by JackBoosY** [comment]
Pinging @Neumann-A  for response. Is work still being done for this PR?

**Comment by Neumann-A** [comment]
need #11072 merged for this

File: ports/cmake/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,83 @@
+set(VCPKG_POLICY_EMPTY_INCLUDE_FOLDER enabled)
+vcpkg_from_gitlab(
+    GITLAB_URL https://gitlab.kitware.com/
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO cmake/cmake
+    REF bd580b7d568493a795a855c0710ba4ad88532485
+    SHA512 73df3c69ba51db22dd87959f7605101d26026ef03d08a9d4fc4017205466b72a394fb4287ba7721c564b976b33df87e4f1dbbb6cd78bb5811a6ec4ddbf386a60
+    HEAD_REF master
+)
+if(EXISTS "${CURRENT_INSTALLED_DIR}/bin/Qt5Core.dll")
```
Comment (cenit) [inline ports/cmake/portfile.cmake:10]: why not a OR of all these conditions?

**Comment by Neumann-A** [comment]
@JackBoosY: Ready but the cmake-gui will probably not work on linux (Ubuntu 20.04) due to requiring a correct fontconfig setup or an update for it within vcpkg

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
After some discussion within the team, here's our current thinking:
1. Vcpkg is still focused on delivering libraries for developers, not general-purpose application distribution.
2. However, this port _is_ useful to test Qt

Therefore, we think the best path is to move this into a secondary location (`scripts/test_ports/cmake`) which can be passed during CI as `--overlay-ports=scripts/test_ports`. This enables the testing coverage granted by building cmake, but avoids introducing false expectations for users that they should `vcpkg install cmake` and expect to run the gui+etc out of the installed tree.

**Comment by Neumann-A** [comment]
Would be nice if VCPKG would store overlay locations of installed ports and only error if the location does not exist any more. Otherwise I always need to pass `--overlay-ports=scripts/test_ports` if one if the ports from there is installed.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**